bindings/input_binding.go

```diff
@@ -16,25 +16,29 @@ package bindings
 import (
 	"context"
 	"fmt"
+	"io"
 
 	"github.com/dapr/components-contrib/health"
 )
 
 // InputBinding is the interface to define a binding that triggers on incoming events.
 type InputBinding interface {
 	// Init passes connection and properties metadata to the binding implementation.
-	Init(metadata Metadata) error
+	Init(ctx context.Context, metadata Metadata) error
 	// Read is a method that runs in background and triggers the callback function whenever an event arrives.
 	Read(ctx context.Context, handler Handler) error
+	// Close is a method that closes the connection to the binding. Must be
+	// called when the binding is no longer needed to free up resources.
+	io.Closer
```

Comment:
Actually because looks like this was not added to other components, perhaps worth not adding it officially here

Reply:
Happy to remove, though I fear that if this is not made part of the interface callers are less likely to call `Close`, or easier to miss because the author didn't attempt to do the type assertion. Ensuring everything is cleaned up and resources released is quite important for things like hot reloading, if we want to do that in future.

Reply:
Ok, let's leave it.

PS:

> I  fear that if this is not made part of the interface callers are less likely to call Close

The only caller right now is dapr/dapr and in just one spot :) Although we are aware that there's _at least one_ person out there using dapr/components-contrib as a library, it's not officially supported or sanctioned.